From 63d5b4ec7a5853bbbae2e1963d406bb09e77dd60 Mon Sep 17 00:00:00 2001
From: Valentin Popa <valentin.popa@intel.com>
Date: Thu, 27 Feb 2014 10:32:17 +0200
Subject: [PATCH] Add MESA_EGL_NO_X11_HEADERS to defines

MESA_EGL_NO_X11_HEADERS is used in eglplatform.h
but the preprocessor doesn't know about it.
Define MESA_EGL_NO_X11_HEADERS for builds without
x11 in DISTRO_FEATURES.

Backport from 9.2.5.

Upstream-Status: Backport

Signed-off-by: Valentin Popa <valentin.popa@intel.com>
---
 configure.ac | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/configure.ac b/configure.ac
index 2acf579..a1a8487 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1568,6 +1568,14 @@ AC_SUBST([EGL_NATIVE_PLATFORM])
 AC_SUBST([EGL_PLATFORMS])
 AC_SUBST([EGL_CFLAGS])
 
+# If we don't have the X11 platform, set this define so we don't try
+# to include
+# the X11 headers.
+if ! echo "$egl_platforms" | grep -q 'x11'; then
+    DEFINES="$DEFINES -DMESA_EGL_NO_X11_HEADERS"
+    GL_PC_CFLAGS="$GL_PC_CFLAGS -DMESA_EGL_NO_X11_HEADERS"
+fi
+
 AC_ARG_WITH([egl-driver-dir],
     [AS_HELP_STRING([--with-egl-driver-dir=DIR],
                     [directory for EGL drivers [[default=${libdir}/egl]]])],
-- 
1.8.3.2

